examples/agora-vote/src/agora-vote.test.ts

```diff
@@ -0,0 +1,263 @@
+    const eventActionPayload = {
+      actionClaimant: {
+        chainid: optimism.id,
+        signatureType: SignatureType.EVENT,
+        signature: selector, // VoteCast(address,uint256,uint8,uint256,string) event signature
+        fieldIndex: 0, // Targeting the 'voter' address
+        targetContract: targetContract, // The Agora vote contract we're monitoring
+      },
+      actionSteps: [eventActionStepOne, eventActionStepTwo],
+    };
+    // Initialize EventAction with the custom payload
+    const eventAction = core.EventAction(eventActionPayload);
+    // Create the boost using the custom EventAction
+    await core.createBoost({
+      protocolFee: 1n,
+      referralFee: 2n,
+      maxParticipants: 100n,
+      budget: budget, // Use the ManagedBudget
+      action: eventAction, // Pass the manually created EventAction
+      validator: core.SignerValidator({
+        signers: [owner, trustedSigner.account], // Whichever account we're going to sign with needs to be a signer
+        validatorCaller: fixtures.core.assertValidAddress(), // Only core should be calling into the validate otherwise it's possible to burn signatures
+      }),
+      allowList: core.SimpleAllowList({
+        owner: owner,
+        allowed: [owner],
+      }),
+      incentives: [
+        core.ERC20VariableIncentive({
+          asset: erc20.assertValidAddress(),
+          reward: parseEther('0.1'),
+          limit: parseEther('1'),
+        }),
+      ],
+    });
+
+    // Make sure the boost was created as expected
+    expect(await core.getBoostCount()).toBe(1n);
+    const boost = await core.getBoost(0n);
+    const action = boost.action;
+    expect(action).toBeDefined();
+
+    // Use viem to send the transaction from the impersonated account
+    await walletClient.impersonateAccount({
+      address: boostImpostor,
+    });
+    await walletClient.setBalance({
+      address: boostImpostor,
+      value: parseEther('10'),
+    });
+
+    const txHash = await walletClient.sendTransaction({
+      data: inputData,
+      account: boostImpostor,
+      to: targetContract,
+    });
+    const txReceipt = await walletClient.getTransactionReceipt({
+      hash: txHash,
+    });
+    await walletClient.mine({ blocks: 1 });
+
+    // Make sure that the transaction was sent as expected and validates the action
+    expect(txHash).toBeDefined();
+
+    const event = (events.abi as Record<Hex, AbiEvent>)[selector] as AbiEvent;
+
+    if (!event) {
+      throw new Error(`No known ABI for given event signature: ${selector}`);
+    }
+
+    const logs = await walletClient.getLogs({
+      address: targetContract,
+      event,
+      fromBlock: OPT_CHAIN_BLOCK,
+      toBlock: 'latest',
+    });
+    const validation = await action.validateActionSteps({
+      logs,
+    });
+    expect(validation).toBe(true);
+
+    const amountOfVotes = await walletClient.readContract({
+      address: '0xcdf27f107725988f2261ce2256bdfcde8b382b10',
+      abi: [
```

Comment:
if we add the getVotes function to packages/signatures/manifests/functions.json, we could pull this AbiItem off `functions.abi[functions.selectors['getVotes(...)']]`, maybe we can also index the AbiItems with the name as well to make that simpler